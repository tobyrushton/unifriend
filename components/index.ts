export * from './providers'
export * from './screens'
