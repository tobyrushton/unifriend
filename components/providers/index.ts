export * from './LoadingProvider'
export * from './UserProvider'
export * from './NotificationProvider'
