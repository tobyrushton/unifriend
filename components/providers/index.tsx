export * from './LoadingProvider'
export * from './UserProvider'