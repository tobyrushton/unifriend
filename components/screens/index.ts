export * from './LoadingScreen'
export * from './AuthScreen'
export * from './ConfirmEmailScreen'
export * from './ForgottenPasswordScreen'
