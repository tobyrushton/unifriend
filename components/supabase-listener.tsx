'use client'

import { useRouter } from 'next/navigation'
import { useEffect, FC } from 'react'
import { supabase } from '../lib/supabase'

export const SupabaseListener: FC<{ accessToken?: string }> = ({
    accessToken,
}) => {
    const router = useRouter()

    useEffect(() => {
        supabase.auth.onAuthStateChange((_event, session) => {
            if (session?.access_token !== accessToken) {
                router.refresh()
            }
        })
    }, [accessToken, router])

    return null
}
