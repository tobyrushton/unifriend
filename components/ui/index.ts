'use client'

export * from './Logo'
export * from './Text'
export * from './Button'
export * from './Input'
export * from './Exit'
export * from './Notification'
export * from './Navbar'
export * from './DropDown'
export * from './ProfilePicture'
export * from './Toggle'
export * from './Message'
export * from './User'
