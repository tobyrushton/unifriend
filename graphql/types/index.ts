export * from './User'
export * from './Friends'
export * from './Messages'
