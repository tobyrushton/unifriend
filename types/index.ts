export * from './User'
export * from './hooks'
export * from './providers'
export * from './Messages'
export * from './Friends'