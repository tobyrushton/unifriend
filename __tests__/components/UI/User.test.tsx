import { render, screen, act, waitFor } from '@testing-library/react'
import { MockedProvider } from '@apollo/client/testing'
import { User } from '../../../components'
import { UserObjectWithID } from '../../../types'
import { ProviderStack } from '../../__helpers__/ProviderStack'
import '@testing-library/jest-dom'

const bio = // eslint-disable-next-line max-len
    'Lorem ipsum dolor sit amet, consectetuer adipiscing elit. Aenean commodo ligula eget dolor. Aenean massa. Cum sociis natoque penatibus et magnis dis parturient montes, nascetur ridiculus mus. Donec quam felis, ultricies nec, pellentesque eu, pretium quis,.'

const user: UserObjectWithID = {
    id: '123',
    firstName: 'Toby',
    lastName: 'Rushton',
    email: '',
    course: 'Computer Science',
    birthday: '2005-06-12',
    username: 'tobyrushton',
    bio,
    university: 'University of Test',
}

// This is a test for the User component
describe('User', () => {
    it('should render', () => {
        const { container } = render(
            <User user={user} getNewUser={jest.fn()} />
        )
        expect(container).toMatchSnapshot()
    })
    it('should render correct user information', () => {
        render(<User user={user} getNewUser={jest.fn()} />)
        expect(screen.getByText('Toby Rushton')).toBeInTheDocument()
        expect(screen.getByText('ðŸ“š Computer Science')).toBeInTheDocument()
        expect(screen.getByText('ðŸŽ“ University of Test')).toBeInTheDocument()
        expect(screen.getByText('ðŸŽ‚ 17 years old')).toBeInTheDocument()
        expect(screen.getByText(bio)).toBeInTheDocument()
    })
    it('should render popup', async () => {
        const { getByText } = await act(async () =>
            render(
                <MockedProvider>
                    <ProviderStack>
                        <User user={user} getNewUser={jest.fn()} />
                    </ProviderStack>
                </MockedProvider>
            )
        )
        const button = getByText('Toby Rushton')
        act(() => button.click())
        expect(getByText('Send friend request?')).toBeInTheDocument()
    })
    it('should render popup and send friend request', async () => {
        const { getByText } = await act(async () =>
            render(
                <MockedProvider>
                    <ProviderStack>
                        <User user={user} getNewUser={jest.fn()} />
                    </ProviderStack>
                </MockedProvider>
            )
        )
        const button = getByText('Toby Rushton')
        act(() => button.click())
        expect(getByText('Send friend request?')).toBeInTheDocument()
        const sendButton = getByText('Send')
        act(() => sendButton.click())
        waitFor(() =>
            expect(getByText('Friend request sent')).toBeInTheDocument()
        )
    })
})
