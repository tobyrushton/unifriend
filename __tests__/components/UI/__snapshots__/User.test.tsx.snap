// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`User should render 1`] = `
<div>
  <div
    class="user"
  >
    <div
      class="userHeader"
    >
      <a
        href="/a/profile/123"
      >
        <img
          alt="Profile Picture"
          class="profilePicture"
          data-nimg="1"
          decoding="async"
          height="75"
          loading="lazy"
          src="http://localhost/_next/image?url=%2FProfile-picture.png&w=256&q=75"
          srcset="/_next/image?url=%2FProfile-picture.png&w=96&q=75 1x, /_next/image?url=%2FProfile-picture.png&w=256&q=75 2x"
          style="color: transparent; background-size: cover; background-position: 50% 50%; background-repeat: no-repeat;"
          width="75"
        />
      </a>
      <div
        class="userHeaderText"
      >
        <div
          role="button"
          tabindex="0"
        >
          <h1
            class="fontText extraLarge clickable left"
          >
            Toby
             
            Rushton
          </h1>
        </div>
        <p
          class="fontText   left"
        >
          ðŸ“š 
          Computer Science
        </p>
        <p
          class="fontText   left"
        >
          ðŸŽ“ 
          University of Test
        </p>
        <p
          class="fontText   left"
        >
          ðŸŽ‚ 
          17
           years old
        </p>
      </div>
    </div>
    <div
      class="detailsContainer"
    >
      <p
        class="fontText   left"
      >
        Lorem ipsum dolor sit amet, consectetuer adipiscing elit. Aenean commodo ligula eget dolor. Aenean massa. Cum sociis natoque penatibus et magnis dis parturient montes, nascetur ridiculus mus. Donec quam felis, ultricies nec, pellentesque eu, pretium quis,.
      </p>
    </div>
  </div>
</div>
`;
